server/src/main/java/com/testsigma/controller/TestCaseResultsController.java

```diff
@@ -52,6 +52,9 @@ public Page<TestCaseResultDTO> index(TestCaseResultSpecificationsBuilder builder
   public TestCaseResultDTO show(@PathVariable(value = "id") Long id) throws ResourceNotFoundException {
     log.info("Request /test_case_results/" + id);
     TestCaseResult testCaseResult = testCaseResultService.find(id);
+    if(testCaseResult.getMessage().contains("architecture is unsupported")) {
+      testCaseResult.setMessage("Architecture is unsupported for the selected application. Please select the correct application file or change the device architecture");
```

Comment:
If this message is specific to Emulator, we can change the message to include Emulator/simulator related details
Ex(include): If running on Simulator, try uploading a simulator buid

Reply:
This is not specific to simulators and hence should be a generic message. But we can definitely include the line you suggested